src/yacs-web/src/pages/Main.vue

```diff
@@ -1,8 +1,8 @@
 <template>
   <b-container fluid class="pt-3">
+    <Header class='mb-3'></Header>
```

Comment:
I think you need to register a listener for the `@new-semester` event here

Reply:
How would I add a listener there? What am I listening for?

Reply:
So in `Header.vue` you're `watch`ing `selectedsemester` and when the value changes you call `this.$emit('new-semester', val)`. So, when the value of `selectedsemester` changes, `<Header>` will emit the event `'new-semester'` with the new value. So to update `currentSemester` in `Main.vue`, just add `@new-semester="newSemester"` to your `<Header>` tag. An example of this is `<CourseList @addCourse="addCourse" />